fe/fe-core/src/main/java/com/starrocks/alter/OnlineOptimizeJobV2.java

```diff
@@ -0,0 +1,814 @@
+        if (db != null) {
+            Locker locker = new Locker();
+            locker.lockDatabase(db, LockType.WRITE);
+            try {
+                OlapTable tbl = (OlapTable) db.getTable(tableId);
+                if (tbl != null) {
+                    onReplayFinished(replayedJob, tbl);
+                }
+            } finally {
+                locker.unLockDatabase(db, LockType.WRITE);
+            }
+        }
+
+        this.jobState = JobState.FINISHED;
+        this.finishedTimeMs = replayedJob.finishedTimeMs;
+
+        LOG.info("replay finished optimize job: {}", jobId);
+    }
+
+    /**
+     * Replay job in CANCELLED state.
+     */
+    private void replayCancelled(OnlineOptimizeJobV2 replayedJob) {
+        cancelInternal();
+        this.jobState = JobState.CANCELLED;
+        this.finishedTimeMs = replayedJob.finishedTimeMs;
+        this.errMsg = replayedJob.errMsg;
+        LOG.info("replay cancelled optimize job: {}", jobId);
+    }
+
+    @Override
+    public void replay(AlterJobV2 replayedJob) {
+        OnlineOptimizeJobV2 replayedOptimizeJob = (OnlineOptimizeJobV2) replayedJob;
+        switch (replayedJob.jobState) {
+            case PENDING:
+                replayPending(replayedOptimizeJob);
+                break;
+            case WAITING_TXN:
+                replayWaitingTxn(replayedOptimizeJob);
+                break;
+            case FINISHED:
+                replayFinished(replayedOptimizeJob);
+                break;
+            case CANCELLED:
+                replayCancelled(replayedOptimizeJob);
+                break;
+            default:
+                break;
+        }
+    }
+
+    @Override
+    protected void getInfo(List<List<Comparable>> infos) {
+        List<Comparable> info = Lists.newArrayList();
+        info.add(jobId);
+        info.add(tableName);
+        info.add(TimeUtils.longToTimeString(createTimeMs));
+        info.add(TimeUtils.longToTimeString(finishedTimeMs));
+        info.add(optimizeOperation != null ? optimizeOperation : "");
+        info.add(watershedTxnId);
+        info.add(jobState.name());
+        info.add(errMsg);
+        // progress
+        info.add(progress);
+        info.add(timeoutMs / 1000);
+        infos.add(info);
+    }
+
+    public void setJobState(JobState jobState) {
+        this.jobState = jobState;
+    }
+
+    @Override
+    public void write(DataOutput out) throws IOException {
+        String json = GsonUtils.GSON.toJson(this, OnlineOptimizeJobV2.class);
+        Text.writeString(out, json);
+    }
+
+    @Override
+    public void gsonPostProcess() throws IOException {
+        if (jobState != JobState.PENDING) {
+            return;
+        }
+    }
+
+    @Override
+    public Optional<Long> getTransactionId() {
+        return watershedTxnId < 0 ? Optional.empty() : Optional.of(watershedTxnId);
+    }
+
+    protected void executeSql(String sql) throws Exception {
```

Comment:
why not use `TaskManager`?

Reply:
1. task framework use fixed thread pool to execute sql, so that rewrite task will be queued if too many tasks(mv task). during the rewrite task, the partition is double write, it's a big overhead so that we want to minimize the time of double write.
2. no need to implement version overwrite syntax for standard sql